### DISCUSSION TITLE
Fire arrows won't prime TNT outside of claims

### DISCUSSION THREAD

**Discussion by MithrandirCraft** [GriefPrevention/GriefPrevention#2042] [Ideas]
### What happened:
Fire arrows don't prime TNT outside of claims. They will, however, prime TNT within a claim.

### What was expected:
Fire arrows should prim TNT both inside and outside of claims.

### Steps to reproduce:
__This may include a build schematic, a video, or detailed instructions to help reconstruct the issue__
1. Enchant your bow with Flame I
2. Place a TNT on the ground
3. Make sure you are outside a claim
4. Shoot the tnt. The TNT will not prime itself for explosion.

Comments: Any other valid vanilla source for priming a TNT will function perfectly.

### Server and GriefPrevention version:
Paste of `/version`: This server is running Paper version git-Paper-232 (MC: 1.17.1) (Implementing API version 1.17.1-R0.1-SNAPSHOT) (Git: 7d10cde)
You are 34 version(s) behind
Paste of `/version GriefPrevention`: GriefPrevention version ed57f0b

### GriefPrevention config.yml

```
GriefPrevention:
  SeaLevelOverrides:
    Survival: -1
    Survival_nether: -1
    Survival_the_end: -1
  Claims:
    Mode:
      Survival: Survival
      Survival_nether: Survival
      Survival_the_end: Survival
    PreventGlobalMonsterEggs: true
    PreventTheft: true
    ProtectCreatures: true
    PreventButtonsSwitches: true
    LockWoodenDoors: true
    LockTrapDoors: true
    LockFenceGates: true
    EnderPearlsRequireAccessTrust: true
    RaidTriggersRequireBuildTrust: true
    ProtectHorses: true
    ProtectDonkeys: true
    ProtectLlamas: true
    InitialBlocks: 3000
    Claim Blocks Accrued Per Hour:
      Default: 250
    Max Accrued Claim Blocks:
      Default: 13000
    Accrued Idle Threshold: 0
    AccruedIdlePercent: 0
    AbandonReturnRatio: 1.0
    AutomaticNewPlayerClaimsRadius: -1
    AutomaticNewPlayerClaimsRadiusMinimum: 0
    ExtendIntoGroundDistance: 5
    MinimumWidth: 5
    MinimumArea: 100
    MaximumDepth: 0
    InvestigationTool: GOLDEN_AXE
    ModificationTool: GOLDEN_SHOVEL
    Expiration:
      ChestClaimDays: 7
      UnusedClaimDays: 30
      AllClaims:
        DaysInactive: 30
        ExceptWhenOwnerHasTotalClaimBlocks: 999999
        ExceptWhenOwnerHasBonusClaimBlocks: 999999
      AutomaticNatureRestoration:
        SurvivalWorlds: false
    AllowTrappedInAdminClaims: false
    MaximumNumberOfClaimsPerPlayer: 20
    CreationRequiresWorldGuardBuildPermission: true
    VillagerTradingRequiresPermission: false
    CommandsRequiringAccessTrust: /sethome;/esethome;/createhome;/ecreatehome "/marry
      sethome"]'
    DeliverManuals: false
    ManualDeliveryDelaySeconds: 30
    RavagersBreakBlocks: true
    FireSpreadsInClaims: false
    FireDamagesInClaims: false
    LecternReadingRequiresAccessTrust: true
  Spam:
    Enabled: false
    LoginCooldownSeconds: 5
    LoginLogoutNotificationsPerMinute: 5
    ChatSlashCommands: /me;/tell;/global;/local;/w;/msg;/r;/t
    WhisperSlashCommands: /tell;/pm;/r;/w;/whisper;/t;/msg
    WarningMessage: No spamees porfavor.
    BanOffenders: true
    BanMessage: Ban por spam.
    AllowedIpAddresses: 1.2.3.4; 5.6.7.8
    DeathMessageCooldownSeconds: 60
    Logout Message Delay In Seconds: 0
  PvP:
    RulesEnabledInWorld:
      Survival: true
      Survival_nether: true
      Survival_the_end: true
    ProtectFreshSpawns: false
    PunishLogout: true
    CombatTimeoutSeconds: 10
    AllowCombatItemDrop: false
    BlockedSlashCommands: /server
    ProtectPlayersInLandClaims:
      PlayerOwnedClaims: true
      AdministrativeClaims: true
      AdministrativeSubdivisions: true
    AllowLavaDumpingNearOtherPlayers:
      PvPWorlds: true
      NonPvPWorlds: false
    AllowFlintAndSteelNearOtherPlayers:
      PvPWorlds: true
      NonPvPWorlds: false
    ProtectPetsOutsideLandClaims: false
  Economy:
    ClaimBlocksMaxBonus: 0
    ClaimBlocksPurchaseCost: 0.0
    ClaimBlocksSellValue: 0.0
  ProtectItemsDroppedOnDeath:
    PvPWorlds: true
    NonPvPWorlds: true
  BlockLandClaimExplosions: true
  BlockSurfaceCreeperExplosions: true
  BlockSurfaceOtherExplosions: false
  LimitSkyTrees: true
  LimitTreeGrowth: false
  PistonMovement: CLAIMS_ONLY
  PistonExplosionSound: true
  FireSpreads: false
  FireDestroys: false
  AdminsGetWhispers: true
  AdminsGetSignNotifications: false
  SmartBan: true
  Mute New Players Using Banned Words: true
  MaxPlayersPerIpAddress: 30
  SilenceBans: true
  Siege:
    Worlds: []
    BreakableBlocks:
    - GRASS_BLOCK
    - DIRT
    - COBBLESTONE
    - OAK_PLANKS
    - SPRUCE_PLANKS
    - BIRCH_PLANKS
    - JUNGLE_PLANKS
    - ACACIA_PLANKS
    - DARK_OAK_PLANKS
    - SAND
    - GRAVEL
    - GLASS
    - GRASS
    - FERN
    - DEAD_BUSH
    - WHITE_WOOL
    - ORANGE_WOOL
    - MAGENTA_WOOL
    - LIGHT_BLUE_WOOL
    - YELLOW_WOOL
    - LIME_WOOL
    - PINK_WOOL
    - GRAY_WOOL
    - LIGHT_GRAY_WOOL
    - CYAN_WOOL
    - PURPLE_WOOL
    - BLUE_WOOL
    - BROWN_WOOL
    - GREEN_WOOL
    - RED_WOOL
    - BLACK_WOOL
    - SNOW
    - GLASS_PANE
    DoorsOpenDelayInSeconds: 300
    CooldownEndInMinutes: 60
  EndermenMoveBlocks: false
  SilverfishBreakBlocks: false
  CreaturesTrampleCrops: false
  RabbitsEatCrops: true
  HardModeZombiesBreakDoors: false
  Database:
    URL: *
    UserName: *
    Password: *
  UseBanCommand: false
  BanCommandPattern: ban %name% %reason%
  Advanced:
    fixNegativeClaimblockAmounts: true
    ClaimExpirationCheckRate: 60
    OfflinePlayer_cache_days: 90
  Abridged Logs:
    Days To Keep: 7
    Included Entry Types:
      Social Activity: true
      Suspicious Activity: false
      Administrative Activity: true
      Debug: true
      Muted Chat Messages: true

```
### Debugging steps attempted

- [ ] (If applicable) I attempted running the server without GriefPrevention installed
  - [ ] (If applicable) The problem does _not_ occur if GriefPrevention is removed from the server.
- [ ] I attempted running only GriefPrevention on the server
  - [ ] Issue still occurs even if GriefPrevention is the only plugin running
- [ ] I attempted testing for the issue on a new server
  - [ ] Issue still occurs even after testing on a new server
- [ ] I attempted testing for the issue with the vanilla client (no mods)
  - [ ] Issue still occurs even after testing with the vanilla Minecraft client

**Comment by Jikoo** [comment]
GP has always used the fire spread rules for fire arrows in wilderness as far as I'm aware, that's why you see this behavior.
https://github.com/TechFortress/GriefPrevention/blob/e6ce386bcff21957af225b5d01e46e4b54d5ad38/src/main/java/me/ryanhamshire/GriefPrevention/EntityEventHandler.java#L274-L280
While it doesn't really make sense to me and I'm good with its removal, it is expected behavior.

/e: I appear to be unable to convert issues to discussions any more (dang it GitHub, changing the way permissions work) so here we are.

**Reply by RoboMWM** [reply]
I'm trying to find the original commit/piece of code for this, blame goes back to you, not sure if you know where.

**Reply by RoboMWM** [reply]
https://github.com/TechFortress/GriefPrevention/commit/a3d0aecaf04984103625cb60a102eb46bf4ecb57 hmmm.....

**Reply by RoboMWM** [reply]
https://github.com/TechFortress/GriefPrevention/pull/931 ok so this old behavior still exists somewhere. Would this be a plugin thing, or was that a vanilla thing?

**Reply by RoboMWM** [reply]
or probs old spigot/craftbukkit handling of fire arrows

**Reply by Jikoo** [reply]
Yeah, it looks like I introduced it in #1424 which I can only assume happened because I did consecutive PRs for the same area of code so #1423 didn't get a proper review, where I would have presumably been told it was a silly change.

**Reply by Jikoo** [reply]
Oh, you're right, it was already in. Honestly not sure, it may have been a byproduct of another event handler? Or I misunderstood what GP was trying to do.

**Reply by RoboMWM** [reply]
That or perhaps that's how spigot handled fire arrows (fire-related event?) before making a change to how they were handled. I can't really recall lol.

**Reply by Jikoo** [reply]
You can actually see the OG handling the line below that PR's diff: https://github.com/TechFortress/GriefPrevention/pull/931/files#diff-64d21d4ede44ab4369b475deda5f1376a61eea360afead70ebf72359b2a4ab1dR723

**Reply by Jikoo** [reply]
Okay, yeah, I've done the traceback on it all the way through, it's been there since the beginning. I guess the question is, when did flaming arrows start igniting TNT? Because GP has been blocking that since the get-go, and it's possible that was a new feature accidentally blocked since before Mojang even added it.
Global fire spread has been affecting it since 3.3.2, but it has been blocked entirely since 3.2 at minimum, at which point we have to jump files again.
https://github.com/TechFortress/GriefPrevention/blame/1f6223526b74cc9e1735e5fc2557a4cc8c7c044e/src/me/ryanhamshire/GriefPrevention/BlockEventHandler.java#L502

**Reply by Jikoo** [reply]
Flaming arrows igniting TNT was added in Minecraft 1.3, which was July 2012. GP has in fact just never adjusted to account for it (3.2 was March 2012), and has treated all ignition as something that would cause in-world fire.

**Reply by RoboMWM** [reply]
ah, that makes sense, I wasn't looking for the inverse condition (block any ignition that _isn't_ x). Thanks for the investigation!

**Comment by MithrandirCraft** [comment]
What, seriously? I don't get it. It's not like fire arrows can cause mass destruction of forests. 

I think we should really distinguish the issue from the cause in this case: The issue being fire spread mass destroying vegetation and wood structures, and the cause being fire blocks, not fire arrows .-.

GP back at it again with it's weird internal stuff. :'D

**Comment by zefir-git** [comment]
Is there any solution/workaround to this?

**Comment by RoboMWM** [comment]
I wonder how old this behavior is. I also find it to be a weird niche case - maybe he was thinking more along the lines of fireballs? I'm curious what the reasoning behind that is.
